runner.go

```diff
@@ -24,11 +32,60 @@ type Runner struct {
 	startedc chan startInfo
 
 	params RunnerParams
+
+	// isDying is maintained by the run goroutine
+	// When it is dying (whether as a result of being killed or due to a
+	// fatal error), all existing workers are killed, no new workers
+	// will be started, and the loop will exit when all existing
+	// workers have stopped.
+	isDying bool
+
+	// finalError holds the error that will be returned
+	// when the runner finally exits.
+	finalError error
```

Comment:
Added a comment to finalError - isDying already says "isDying is maintained by the run goroutine", and I *think* that as it's not inside a mutex, it should be clear that "maintained by" implies ownership and hence that it shouldn't be modified outside of that goroutine.

I'm open to alternative comment wordings.